src/pipeline.ml

```diff
@@ -46,28 +45,51 @@ let install_package_archive opam_image =
   copy ~chown:"0:0" ~from:"archive" ~src:["/home/opam/opam-repository/cache"] ~dst:"/cache" ()
 
 (* Generate a Dockerfile to install OCaml compiler [switch] in [opam_image]. *)
-let install_compiler_df ~arch ~switch opam_image =
-  let switch_name = Ocaml_version.to_string (Ocaml_version.with_just_major_and_minor switch) in
-  let (package_name, package_version) = Ocaml_version.Opam.V2.package switch in
-  let additional_packages = Ocaml_version.Opam.V2.additional_packages switch in
+let install_compiler_df ~os_family ~arch ~switch ?windows_port opam_image =
+  let switch_name = Ocaml_version.(with_just_major_and_minor switch |> to_string) in
+  let (package_name, package_version) =
+    match windows_port with
+    | None -> Ocaml_version.Opam.V2.package switch
+    | Some port -> Dockerfile_windows.ocaml_for_windows_package_exn ~switch ~arch ~port
+  in
   let open Dockerfile in
-  let personality = if Ocaml_version.arch_is_32bit arch then shell ["/usr/bin/linux32"; "/bin/sh"; "-c"] else empty in
+  let personality = Dockerfile_distro.personality os_family arch in
+  let run, run_no_opam, depext, etp_exec =
```

Comment:
`etp_exec` is a bit cryptic - just `entrypoint`?

Reply:
Changed to `opam_exec`, `entrypoint` would shadow `Dockerfile.entrypoint`.